python/seldon_core/user_model.py

```diff
@@ -1,15 +1,24 @@
 from seldon_core.metrics import validate_metrics
 from seldon_core.flask_utils import SeldonMicroserviceException
 import json
-from typing import Dict, List, Union, Iterable, Callable, Optional
+from typing import Dict, List, Union, Iterable
 import numpy as np
 from seldon_core.proto import prediction_pb2
+from seldon_core.metrics import SeldonMetrics
 import inspect
 import logging
+import os
 
 logger = logging.getLogger(__name__)
 
 
+ENV_CLIENT_GETS_METRICS = "INCLUDE_METRICS_IN_CLIENT_RESPONSE"
+
+
+def return_metrics_to_client():
+    return os.environ.get(ENV_CLIENT_GETS_METRICS, "true").lower() == "true"
```

Comment:
We'd want to avoid reading from environment for every request as that would be qutie an expensive IO

Reply:
Then I think I had it done better a commit or two ago, if you think it's better I can revert.

Reply:
This way was cleaner for unit tests.

Reply:
Oh I see, I guess because you can inject  os.environ, yeah I definitely see it being cleaner - although the main reason for this PR is for efficiency so probably best to avoid reading env as it's significantly slower. The way that we do it in Golang is by this being loaded at the top of the file and override the env variableat the beginning of the test.

Reply:
I reverted to the previous approach, should be better for performance now!